include/ttmlir/Dialect/TT/IR/TTOpsEnums.td

```diff
@@ -72,6 +72,26 @@ def TT_MemorySpace : I32EnumAttr<"MemorySpace", "TT MemorySpace",
   let cppNamespace = "::mlir::tt";
 }
 
+def TT_UndefLayout : I32EnumAttrCase<"UndefLayout", 0, "undef_layout">; // For host tensors
```

Comment:
Maybe we should call this None to match flatbuffer enum definition.  Also this comment "For host tensors" isn't strictly true, it'll also always be None/Undef for direct to metal path.